video/out/vo.c

```diff
@@ -1076,14 +1076,15 @@ static void do_redraw(struct vo *vo)
 {
     struct vo_internal *in = vo->in;
 
-    if (!vo->config_ok || (vo->driver->caps & VO_CAP_NORETAIN) ||
-        (vo->driver->caps & VO_CAP_UNTIMED))
+    if (!vo->config_ok || (vo->driver->caps & VO_CAP_NORETAIN))
         return;
 
     mp_mutex_lock(&in->lock);
     in->request_redraw = false;
     bool full_redraw = in->dropped_frame;
-    struct vo_frame *frame = vo_frame_ref(in->current_frame);
+    struct vo_frame *frame = NULL;
+    if (!(vo->driver->caps & VO_CAP_UNTIMED))
+        frame = vo_frame_ref(in->current_frame);
```

Comment:
This does a use after free for `vo_image` and so on. The whole reason why I early exited it.

Reply:
What do you suggest? We cannot spin vo in paused state, I think.

Reply:
The problem is that `in->request_redraw` needs to always be set to false. This should mean that `VO_CAP_NORETAIN` has also always been broken and was spinning. Early returning if `!vo->config_ok` should be OK though.

Reply:
changed